Dockerfile

```diff
@@ -93,4 +93,10 @@ ENV TEST_DEBUG=0
 
 WORKDIR /root/lightning-integration
 COPY Makefile /root/lightning-integration/Makefile
+
+RUN make src/eclair bin/eclair.jar
+RUN make src/ptarmigan bin/ptarmd
+RUN make src/lightning bin/lightningd
+RUN make src/lnd bin/lnd
```

Comment:
No, this makes the image larger and compiles in the current `master` version when the image is being built. The docker image should be only the test runner, not commit to a specific version.

Reply:
If this is desired, please use derived images instead that can indeed serve as reproducible environments.

Reply:
I understand that the image will be bigger, but I thought that the image would turn just as big when running CMD.
Regarding committing to a version, I don't understand how that would be the case.

If we run ```make update clients``` in the dockerfile some version will e built.
Then if after some time the same image is run again, then ```make update clients``` will be run again, since they're still included in the CMD. The built for the new version will not be cached, just like now they aren't cached, and from that point, having cached anything will be mostly pointless and an extra cost in image size, I guess.

If the extra image size alone is enough to nack it, then I will just close it and try to get the other commits merged in another PR (I can always maintain this simple commit locally for convenience), but I don't see how we're committing to a specific version here.

Reply:
Yes, but the builds in the docker image will not be clean, and may contain out of date artefacts that have not been correctly built when updating (it's happened before for c-lightning builds at least). That is my major concern, since the results would differ from what a clean build looks like.

Reply:
Fair enough, I will just rewrite the one commit yet or something similar locally when needed to save some time caching things. Not a big deal.
I'm going to close this one and open another one with the change to the makefile and readme.